renku/domain_model/template.py

```diff
@@ -148,40 +149,64 @@ def get_raw_content(self) -> List[Dict]:
         """Return raw manifest file content."""
         return copy.deepcopy(self._content)
 
-    def validate(self):
+    def validate(self, manifest_only: bool = False) -> List[str]:
         """Validate manifest content."""
+        warnings = []
+
         if not self._content:
             raise errors.InvalidTemplateError("Cannot find any valid template in manifest file")
         elif not isinstance(self._content, list):
             raise errors.InvalidTemplateError(f"Invalid manifest content type: '{type(self._content).__name__}'")
 
         # NOTE: First check if required fields exists for creating Template instances
-        for template in self._content:
-            if not isinstance(template, dict):
-                raise errors.InvalidTemplateError(f"Invalid template type: '{type(template).__name__}'")
+        for template_entry in self._content:
+            if not isinstance(template_entry, dict):
+                raise errors.InvalidTemplateError(f"Invalid template type: '{type(template_entry).__name__}'")
 
-            id = template.get("id") or template.get("folder")
+            id = template_entry.get("id") or template_entry.get("folder")
             if not id:
-                raise errors.InvalidTemplateError(f"Template doesn't have an id: '{template}'")
+                raise errors.InvalidTemplateError(f"Template doesn't have an id: '{template_entry}'")
+            if not template_entry.get("folder"):
+                warnings.append(f"Template '{id}' should use 'folder' attribute instead of 'id'.")
 
-            parameters = template.get("parameters") or template.get("variables")
+            parameters = template_entry.get("parameters") or template_entry.get("variables")
             if parameters:
+                if template_entry.get("parameters"):
+                    warnings.append(f"Template '{id}' should use 'variables' instead of 'parameters' in manifest.")
```

Comment:
It's actually the case in all templates. IMO `parameters` is a better name than `variables` but it's ok if we want to keep `variables`. In that case we don't need this check and we should remove `template_entry.get("parameters")`. What do you think?